src/property-filter/index.tsx

```diff
@@ -149,65 +150,75 @@ const PropertyFilter = React.forwardRef(
     };
 
     useImperativeHandle(ref, () => ({ focus: () => inputRef.current?.focus() }), []);
-    const { tokens, operation } = query;
-    const showResults = !!tokens?.length && !disabled && !!countText;
+    const showResults = !!query.tokens?.length && !disabled && !!countText;
     const { addToken, removeToken, setToken, setOperation, removeAllTokens } = getQueryActions(
       query,
       onChange,
       inputRef
     );
     const [filteringText, setFilteringText] = useState<string>('');
 
-    const internalFilteringProperties: readonly InternalFilteringProperty[] = filteringProperties.map(property => {
-      const extendedOperators = (property.operators ?? []).reduce(
-        (acc, operator) => (typeof operator === 'object' ? acc.set(operator.operator, operator) : acc),
-        new Map<PropertyFilterOperator, null | ExtendedOperator<any>>()
-      );
-      return {
-        propertyKey: property.key,
-        propertyLabel: property.propertyLabel ?? '',
-        groupValuesLabel: property.groupValuesLabel ?? '',
-        propertyGroup: property.group,
-        operators: (property.operators ?? []).map(op => (typeof op === 'string' ? op : op.operator)),
-        defaultOperator: property.defaultOperator ?? '=',
-        getValueFormatter: operator => (operator ? extendedOperators.get(operator)?.format ?? null : null),
-        getValueFormRenderer: operator => (operator ? extendedOperators.get(operator)?.form ?? null : null),
-        externalProperty: property,
-      };
-    });
-
-    const propertyByKey = new Map(internalFilteringProperties.map(p => [p.propertyKey, p]));
+    const { internalProperties, internalOptions, internalQuery } = useMemo(() => {
```

Comment:
This memo is useless, because collection hooks produces new array on every render anyway: https://github.com/cloudscape-design/collection-hooks/blob/main/src/utils.ts#L116

Please do not add any `useMemo` until you are really sure it works.